### ISSUE TITLE
IE11 does not identify itself as IE (Prototype.Browser.IE is false)

### ISSUE THREAD

**Issue by jwestbrook** [sstephenson/prototype#120]
[previous lighthouse ticket #3508](https://prototype.lighthouseapp.com/projects/8886/tickets/3508-ie11-support) 
by Arnaud Dvl

---

Actually, the 1.7 version of prototype thinks that IE11 is a Mozilla browser.
You can have a look at this link to see all the removed API:
http://msdn.microsoft.com/en-us/library/ie/bg182625(v=vs.85).aspx

In particular, document.all and attachEvent have been removed.

**Comment by plynchnlm** [comment]
I think the importance of this issue depends on whether this flag is meant to be internal to PrototypeJS, or public for PrototypeJS-using applications to use.  I just tried to use this flag in my application, and had to add in jwestbrook's additional test for IE 11.  The documentation at http://api.prototypejs.org/Prototype/Browser/ makes it look like the IE flag is meant to be public.

**Comment by GreenReaper** [comment]
Yeah, I wanted this to let me detect IE but it didn't, reliably. I got Gecko on IE12/Win10.

There's also Edge to consider, which is detected as WebKit.

**Comment by walterdavis** [comment]
These flags are used internally to replace normal code with work-arounds, so you don't try to do something on Firefox that you need to do on IE 9, since that might cause Firefox to stop working. If IE 11 passes all the tests, then the only reason to test for IE is if you want to know for some reason, not because ordinary Prototype standard code won't work in that browser, and something needs to be worked around.

Walter
